src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -979,6 +979,16 @@ public static bool SequenceEqual<TSource>(this IEnumerable<TSource> first, IEnum
             if (comparer == null) comparer = EqualityComparer<TSource>.Default;
             if (first == null) throw Error.ArgumentNull("first");
             if (second == null) throw Error.ArgumentNull("second");
+
+            if (first == second) return true;
```

Comment:
It's pretty rare to call SequenceEqual with the same object.  And if you're doing so on purpose, then it's almost certainly because you expect it's possible that each enumeration could return different results.  So I think this checks buys us very little and could potentially be problematic, and shouldn't be done.